ts-tests/common/setup/wait-finalized-block.ts

```diff
@@ -0,0 +1,59 @@
+import '@polkadot/api-augment';
+import { ApiPromise, Keyring, WsProvider } from '@polkadot/api';
+import type { VoidFn } from '@polkadot/api/types';
+import type { ISubmittableResult } from '@polkadot/types/types';
+import { loadConfig } from '../utils';
+
+const FINALIZED_BLOCKS_COUNT = 1;
+const TIMEOUT_MIN = 1000 * 60 * 3; // 1min
+
+/**
+ * Connects to the parachain via the config file
+ * and waits for `FINALIZED_BLOCKS_COUNT` blocks to
+ * get finalized.
+ *
+ * It times out after `TIMEOUT_MIN` min.
+ */
+(async () => {
+    const config = loadConfig();
+    let timeout: NodeJS.Timeout;
+    let count = 0;
+    let unsub: VoidFn;
+    let api: ApiPromise;
+
+    const provider = new WsProvider(config.parachain_ws);
+
+    console.log(`Connecting to parachain ${config.parachain_ws}`);
+
+    timeout = global.setTimeout(async () => {
+
+      if (typeof unsub === 'function') unsub();
+
+      if (api) api.disconnect();
+      provider.on('disconnected', () => {
+          console.log(`\nno block production detected after ${TIMEOUT_MIN}min, you might want to check it manually. Quit now`);
+          process.exit(1);
+      });
+    }, TIMEOUT_MIN);
+
+    api = await ApiPromise.create({
+        provider: provider,
+    });
+
+    unsub = await api.rpc.chain.subscribeFinalizedHeads(async (head) => {
+      const blockNumber = head.number.toNumber();
+      console.log(`Parachain finalized block #${blockNumber} with hash ${head.hash}`);
+      count += 1;
+
+      if (blockNumber >= 1 && count >= FINALIZED_BLOCKS_COUNT) {
```

Comment:
I am making it wait for at least block #1. Although while testing it worked if we stopped at block #0. Maybe a race condition of #1 existing when `upgrade-parathread` starts?

Reply:
Yes please, for the script alone we need the finalized block to be non-zero - the test worked at #0 probably because the request (tx) is buffered in the pool.